examples/list-matching-files/main.rs

```diff
@@ -0,0 +1,62 @@
+// Example program illustrating use of shwild to filter by name files found
+// in the executing directory
+
+use std::{
+    env as std_env,
+    fs as std_fs,
+};
+
+
+fn main() {
+    let directory = ".";
+
+    let patterns = {
+        let r = std_env::args().skip(1).collect::<Vec<_>>();
+
+        if r.is_empty() {
+            vec!["*".into()]
+        } else {
+            r
+        }
+    };
+
+    println!("searching in '{directory}' with pattern(s) {patterns:?}");
+
+    match std_fs::read_dir(directory) {
+        Ok(entries) => {
+            // for each file in the directory ...
+            for entry in entries {
+                match entry {
+                    Ok(entry) => {
+                        let path = entry.path();
+                        let path_s = format!("{}", path.display());
+
+                        // ... check against ...
+                        for pattern in &patterns {
+                            // ... each pattern ...
+                            match shwild::matches(pattern, &path_s, 0) {
+                                Ok(is_matched) => {
+                                    if is_matched {
+                                        // ... and print when it matches any one.
+                                        println!("\t{path_s}");
+
+                                        break;
```

Comment:
No labeled `break 'pattern_checking;`?

Reply:
I don't see the need for that. Can you elaborate?